dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,281 @@
+    /// <returns>Text content from audio content.</returns>
+    public async Task<TextContent> GetTextContentAsync(
+        AudioStreamContent content,
+        PromptExecutionSettings? executionSettings = null,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(content);
+
+        string uploadUrl = await this.UploadFileAsync(content.Stream, cancellationToken).ConfigureAwait(false);
+
+        var transcriptId = await this.CreateTranscriptAsync(uploadUrl, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+        var transcript = await this.WaitForTranscriptToProcessAsync(transcriptId, executionSettings, cancellationToken)
+            .ConfigureAwait(false);
+
+        return new TextContent(
+            text: transcript.RootElement.GetProperty("text").GetString(),
+            modelId: null,
+            // TODO: change to typed object when AAI SDK is shipped
+            innerContent: transcript,
+            encoding: Encoding.UTF8,
+            metadata: null
+        );
+    }
+
+    private async Task<string> UploadFileAsync(Stream audioStream, CancellationToken ct)
+    {
+        var url = this.Url("v2/upload");
+        using var content = new StreamContent(audioStream);
+        content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+        using var request = new HttpRequestMessage(HttpMethod.Post, url);
+        this.AddDefaultHeaders(request);
+        request.Content = content;
+        using var response = await this.SendWithSuccessCheckAsync(this.HttpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAndTranslateExceptionAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        return json.RootElement.GetProperty("upload_url").GetString()
+               ?? throw new KernelException("Property 'upload_url' expected but not found.");
+    }
+
+    private async Task<string> CreateTranscriptAsync(
+        string audioUrl,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url("v2/transcript");
+        var jsonRequest = new JsonObject();
+        jsonRequest["audio_url"] = audioUrl;
+        if (executionSettings?.ExtensionData is not null)
+        {
+            foreach (var attribute in executionSettings.ExtensionData)
+            {
+                jsonRequest[attribute.Key] = JsonValue.Create(attribute.Value);
+            }
+        }
+
+        using var request = HttpRequest.CreatePostRequest(url, jsonRequest);
+        this.AddDefaultHeaders(request);
+        using var response = await this.SendWithSuccessCheckAsync(this.HttpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
```

Comment:
```suggestion
        using var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
```
I know disposing response dispose also stream, but good practice is always call dispose if something is disposable.